src/app/shared/login/login.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { AuthService } from '../../core/services/auth.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+  form: FormGroup;
+
+  constructor(private fb: FormBuilder, private authService: AuthService) {
+    this.form = fb.group({
```

Comment:
you should create form in "formclass" and add some validations

Reply:
I agree with validations but why "formclass"? What's wrong with form in component?

Reply:
Done